libs/sdk-ui-dashboard/src/model/events/kpi.ts

```diff
@@ -300,6 +300,44 @@ export const isDashboardKpiWidgetChanged = eventGuard<DashboardKpiWidgetChanged>
 //
 //
 
+/**
+ * This event is emitted after execution of a KPI widget starts.
+ *
+ * @alpha
+ */
+export interface DashboardKpiWidgetExecutionStarted extends IDashboardEvent {
+    readonly type: "GDC.DASH/EVT.KPI_WIDGET.EXECUTION_STARTED";
+    readonly payload: {
+        widgetRef: ObjRef;
```

Comment:
similar points to the above.. some info about metric that is backing the execution, ideally the entire prepared execution would be nice. + some way to reliably match begin-end events.